app/models/hostgroup.rb

```diff
@@ -16,6 +16,7 @@ class Hostgroup < ApplicationRecord
 
   validate :validate_subnet_types
   validates_with SubnetsConsistencyValidator
+  validate :validate_compute_profile, :if => proc { |hostgroup| hostgroup.compute_profile_id.present? }
```

Comment:
I think you could combine also the `changed?` check to handle the case of current profile that the user can't see but isn't changed
```suggestion
  validate :validate_compute_profile, :if => proc { |hostgroup| hostgroup.compute_profile_id_changed? && hostgroup.compute_profile_id.present? }
```